src/shared/fs/templateRegistry.ts

```diff
@@ -72,59 +71,48 @@ export class AsyncCloudFormationTemplateRegistry {
         private readonly instance: CloudFormationTemplateRegistry,
         private readonly asyncSetupFunc: (
             instance: CloudFormationTemplateRegistry,
-            cancelSetup: Timeout
+            cancelSetup: Timeout,
+            onItem?: (item: number) => void
         ) => Promise<CloudFormationTemplateRegistry>
     ) {}
 
     /**
-     * Returns the initial registry instance if setup has completed, otherwise
-     * returning a temporary instance and showing a progress message to the user
-     * to indicate setup is in progress.
+     * Returns the initial registry instance if setup has completed, otherwise returns a temporary
+     * instance and shows a progress message ("Scanning...") until setup is done.
      */
     async getInstance(): Promise<CloudFormationTemplateRegistry> {
         if (this.isSetup) {
             return this.instance
         }
+        if (this.setupPromise) {
+            getLogger().debug('%s: getInstance() requested, still initializing', this.instance.name)
+            return this.setupPromise
```

Comment:
Return early. Before we were calling `setupPromise.then()` multiple times, don't see why that is needed.